Update the RTC Engine to 0.7.0

jerzywilczek opened the pull request.

jerzywilczek committed:
Update the RTC Engine to 0.7.0

varsill commented on lib/videoroom/room.ex:
```diff
@@ -24,11 +24,18 @@ defmodule Videoroom.Room do
     ]
 
+    mock_ip = Application.fetch_env!(:membrane_videoroom_demo, :external_ip)
+    external_ip = if @mix_env == :prod, do: {0, 0, 0, 0}, else: mock_ip
```
`@mix_env` is not an inbuilt tag - you need to define it that way: `@mix_env Mix.env()`, as it was done here: https://github.com/membraneframework/membrane_videoroom/blob/4fc8841510086d8e9f91b71f9262ec1572216bc8/lib/videoroom/room.ex#L17

varsill commented on assets/package-lock.json:
```diff
@@ -4,16 +4,18 @@
   "requires": true,
```
I believe there is an inconsistency with the files package.json files ;< Haven't you installed dependencies with `npm install`?  
<img width="1361" alt="image" src="https://user-images.githubusercontent.com/10851084/205258664-d6c30f4d-b243-4725-b26f-a59cc36aabc5.png">

jerzywilczek replied:
I have and it worked. I'll check this again, perhaps it breaks after deleting `deps` or sth

jerzywilczek replied:
It does work for me. I'll need more info on this, we should talk tomorrow.

varsill commented:
General comment - we should probably explicitly say something about setting up EXTERNAL_IP env before running the application, as the application won't work with Firefox if the env is not set.

jerzywilczek committed:
Define `@mix_env` properly.

jerzywilczek committed:
Correct the error messages when parsing environment variables.

jerzywilczek committed:
Update npm deps

jerzywilczek committed:
Update `package-lock.json` again

jerzywilczek committed:
Add some fixes to enable running the app in vscode dev-containers.

 * the dev container will now publish ports necessary for the TURN to
   work properly and the `ConfigParser` will expect that, so that the
   user doesn't need to specify them manually
 * it will also set a variable signifying that the app is running in
   docker, which allows setting TURN's `external_ip` to `0.0.0.0`
   automatically when running in the dev container

jerzywilczek committed:
Write published ports in a more concise way.

jerzywilczek committed:
Update `fake_turn`

varsill reviewed (approved):
lgtm! 🥇

jerzywilczek committed:
Remove information about removed `template/additional_features` branch.
